Source/FieldSolver/FiniteDifferenceSolver/ComputeCurlA.cpp

```diff
@@ -0,0 +1,289 @@
+    Bfield[1]->setVal(0);
+    Bfield[2]->setVal(0);
+
+    // Loop through the grids, and over the tiles within each grid
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    for ( MFIter mfi(*Afield[0], TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
+        if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers) {
+            amrex::Gpu::synchronize();
+        }
+        auto wt = static_cast<amrex::Real>(amrex::second());
+
+        // Extract field data for this grid/tile
+        Array4<Real> const &Bx = Bfield[0]->array(mfi);
+        Array4<Real> const &By = Bfield[1]->array(mfi);
+        Array4<Real> const &Bz = Bfield[2]->array(mfi);
+        Array4<Real const> const &Ax = Afield[0]->const_array(mfi);
+        Array4<Real const> const &Ay = Afield[1]->const_array(mfi);
+        Array4<Real const> const &Az = Afield[2]->const_array(mfi);
+
+        EB::Covered const& cov_ptr = EB::Covered(mfi, lev);
+
+        // Extract stencil coefficients
+        Real const * const AMREX_RESTRICT coefs_x = m_stencil_coefs_x.dataPtr();
+        auto const n_coefs_x = static_cast<int>(m_stencil_coefs_x.size());
+        Real const * const AMREX_RESTRICT coefs_y = m_stencil_coefs_y.dataPtr();
+        auto const n_coefs_y = static_cast<int>(m_stencil_coefs_y.size());
+        Real const * const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+        auto const n_coefs_z = static_cast<int>(m_stencil_coefs_z.size());
+
+        // Extract tileboxes for which to loop
+        Box const& tbx  = mfi.tilebox(Bfield[0]->ixType().toIntVect());
+        Box const& tby  = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+        Box const& tbz  = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+        // Calculate the curl of A
+        amrex::ParallelFor(tbx, tby, tbz,
+
+            // Bx calculation
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+                // Skip if this cell is fully covered by embedded boundaries
+                if (cov_ptr.isCovered(0, EB::CoverTopology::face, i, j, k)) { return; }
+
+                Bx(i, j, k) =  (
+                    - T_Algo::UpwardDz(Ay, coefs_z, n_coefs_z, i, j, k)
+                    + T_Algo::UpwardDy(Az, coefs_y, n_coefs_y, i, j, k)
+                );
+            },
+
+            // By calculation
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+                // Skip if this cell is fully covered by embedded boundaries
+                if (cov_ptr.isCovered(1, EB::CoverTopology::face, i, j, k)) { return; }
+
+                By(i, j, k) = (
+                    - T_Algo::UpwardDx(Az, coefs_x, n_coefs_x, i, j, k)
+                    + T_Algo::UpwardDz(Ax, coefs_z, n_coefs_z, i, j, k)
+                );
+            },
+
+            // Bz calculation
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+                if (cov_ptr.isCovered(2, EB::CoverTopology::face, i, j, k)) { return; }
```

Comment:
Just to keep things consistent
```suggestion
                // Skip if this cell is fully covered by embedded boundaries
                if (cov_ptr.isCovered(2, EB::CoverTopology::face, i, j, k)) { return; }
```